src/main/java/eu/dissco/core/digitalspecimenprocessor/component/TokenAuthenticator.java

```diff
@@ -0,0 +1,40 @@
+package eu.dissco.core.digitalspecimenprocessor.component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import eu.dissco.core.digitalspecimenprocessor.property.TokenProperties;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+
+@Component
+@RequiredArgsConstructor
+public class TokenAuthenticator {
+
+    private final TokenProperties properties;
+
+    @Qualifier("tokenClient")
+    private final WebClient tokenClient;
+
+    public String getToken() throws ExecutionException {
+        var responseSpec = tokenClient
+                .post()
+                .body(BodyInserters.fromFormData(properties.getFromFormData()))
+                .acceptCharset(StandardCharsets.UTF_8)
+                .retrieve();
+
+        var response = responseSpec.bodyToMono(JsonNode.class);
+        try {
+            var responseJson= response.toFuture().get();
+            return responseJson.get("access_token").asText();
```

Comment:
What if there is no `access_token`? Let's say authentication fails, what happens then?

Reply:
Added some more granular exception handling. I cover two outcomes:
1. if we get a 401 from the keycloak server
2. If we don't get a 401 but there's no "access_token" term

Both cases throw a PidAuthenticationException.

Reply:
Could also throw a PidCreationException, but an authentication error with the keycloak server would likely require human intervention (i.e. updating the key).

Reply:
It could be that Keycloak is down, or that there is some other issue with Keycloak. It would/should mean that we stop the processing service from at least processing new specimen as we cannot provide them with a Handle/DOI